lib/edit-site-page.php

```diff
@@ -155,19 +154,20 @@ function gutenberg_edit_site_init( $hook ) {
 
 		$current_template = gutenberg_find_template_post_and_parts( $template_type );
 		if ( isset( $current_template ) ) {
-			$template_ids[ $current_template['template_post']->post_name ] = $current_template['template_post']->ID;
 			$template_part_ids = $template_part_ids + $current_template['template_part_ids'];
 		}
 	}
 
-	$current_template_id = $template_ids['front-page'];
+	$home_template = gutenberg_find_template_post_and_parts( 'front-page' );
+	if ( isset( $home_template ) ) {
+		$home_template_id = $home_template['template_post']->ID;
+	}
```

Comment:
What happens if there is no template here? That means that not even an `index.html` is provided. Should we create an auto-draft as a fallback?

Reply:
Yeah, maybe. An empty one I guess? I'll implement a fix tomorrow.

Reply:
An empty auto-draft.

Reply:
Upon closer look, I'm no longer totally sure that this is the right solution :thinking: 

When rendering to the frontend, we fall back to an [error message](https://github.com/WordPress/gutenberg/blob/02fe765c41926b6fae6e2a0606c2892e05b5e162/lib/template-loader.php#L354). Creating an emtpy auto-draft here would thus change the frontend output -- I don't think we want that.

`$home_template_id` is really just used in the editor to highlight which template is the home template -- so if there are simply no templates at all, it's probably fine to set it to `null`.

(The issue is also present in current `master`, where it affects both `homeTemplateId` and `templateId`, so I should fix it with a separate PR. What I said above applies to that scenario too, though.)

Reply:
This is needed so people can access the site editor without these templates. Otherwise, they just get errors.

We can just make that error show on the front-end if the template is an empty auto-draft.

Reply:
Exactly